lib/internal/repl/utils.js

```diff
@@ -783,6 +788,43 @@ function isObjectLiteral(code) {
     RegExpPrototypeExec(endsWithSemicolonRegExp, code) === null;
 }
 
+const kContextId = Symbol('contextId');
+
+const path = require('path');
+
+function fixReplRequire(replModule) {
+  try {
+    // Hack for require.resolve("./relative") to work properly.
+    replModule.filename = path.resolve('repl');
+  } catch {
+    // path.resolve('repl') fails when the current working directory has been
+    // deleted.  Fall back to the directory name of the (absolute) executable
+    // path.  It's not really correct but what are the alternatives?
+    const dirname = path.dirname(process.execPath);
+    replModule.filename = path.resolve(dirname, 'repl');
+  }
+
+  // Hack for repl require to work properly with node_modules folders
+  replModule.paths = CJSModule._nodeModulePaths(replModule.filename);
+}
+
+let nextREPLResourceNumber = 1;
+// This prevents v8 code cache from getting confused and using a different
+// cache from a resource of the same name
+function getREPLResourceName() {
+  return `REPL${nextREPLResourceNumber++}`;
+}
+
+const globalBuiltins =
+  new SafeSet(vm.runInNewContext('Object.getOwnPropertyNames(globalThis)'));
+
+const replBuiltinLibsWrapper = {
+  _builtinLibs: ArrayPrototypeFilter(
```

Comment:
It seems fine for this to just be it's own property that gets re-exported instead of being nested here and then accessed again in the public module (I am a bit confused what this is wrapping to get named a "wrapper")

Reply:
I am using a wrapper here because the `_builtinLibs` value can be updated by users in the setters here: https://github.com/nodejs/node/blob/446cedcfb73300d32047c776d8ca7af92b803a48/lib/repl.js#L1399-L1429

And I wanted to make sure that all modules do get the potentially updated value in such case, that's why I used a wrapper to make sure that all get the same object and the `_builtinLibs` fields (updated or not) is shared among them.

Is this the wrong approach? do you see a better/simpler way to achieve this?

PS: note that the setters are deprecated to this should also be temporary, once those are removed in the future the wrapper can be removed anyways

Reply:
Or you can just export methods for get/set it in this module, and let the public module's accessors forward to them? I think that more typically done for values kept in a module's singleton

Reply:
it's used both in the public one and in a few places in the completion module (you can search for `replBuiltinLibsWrapper._builtinLibs` in the `lib/internal/repl/completion.js` file)

that's why I moved it here